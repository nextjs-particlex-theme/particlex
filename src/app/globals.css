@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
    :root {
        --color-white: #fff;
        --color-text-sub: #5c6b72;
        --color-text: #1e3e3f;
        --color-primary: #66afef;
        --color-error: #ef4444;
        --border-color: #D9D9D9FF;
        --z-index-top: 100000;
        --z-index-common: 10000;
        --background-color: #f6f8fa;
    }
}

body {
    background: var(--background-color);
    color: #1e3e3f;
    font: 500 14px Lexend, "Noto Sans SC", sans-serif;
    overflow-x: hidden;
    width: 100%;
}

.link-styled-container a {
    color: var(--color-primary);
    text-decoration: none;
}
a:hover {
    opacity: .8;
}

::-webkit-scrollbar {
    height: 12px;
    width: 12px;
}

::-webkit-scrollbar-thumb {
    background: #8ab5ff linear-gradient(
            45deg,
            #fff6 25%,
            transparent 25%,
            transparent 50%,
            #fff6 50%,
            #fff6 75%,
            transparent 75%,
            transparent
    );
    border: 3px solid #e6efff;
    border-radius: 100px;
}
::-webkit-scrollbar-track {
    background: #e6efff;
    border-radius: 100px;
}
::selection {
    background-color: #3392ff2a;
    color: unset;
}

@keyframes fadeIn {
    from {
        opacity: 0;
        transform: translateY(-20px) scale(1.1);
    }

    to {
        opacity: 1;
        transform: none;
    }
}

/*添加动画*/
#fade-in {
    animation-duration: .8s;
    animation-fill-mode: both;
    animation-name: fadeIn;
}