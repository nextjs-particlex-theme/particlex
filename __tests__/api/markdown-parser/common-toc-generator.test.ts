import { expect, test } from '@jest/globals'
import markdown from './markdown.test.md'
import badHeadingMarkdown from './bad-heading.test.md'
import generateTocByMarkdown from '@/api/markdown-parser/common-toc-generator'

test('Test toc parse', () => {
  const toc = generateTocByMarkdown(markdown)
  expect(toc).toMatchSnapshot()
})


test('Test bad heading generate', () => {
  const toc = generateTocByMarkdown(badHeadingMarkdown)
  expect(toc).toMatchSnapshot()
})