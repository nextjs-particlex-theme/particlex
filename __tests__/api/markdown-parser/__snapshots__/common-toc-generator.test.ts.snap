// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test bad heading generate 1`] = `
[
  {
    "anchor": "#L4",
    "child": [],
    "title": "L4",
  },
  {
    "anchor": "#L3",
    "child": [],
    "title": "L3",
  },
  {
    "anchor": "#L2",
    "child": [],
    "title": "L2",
  },
  {
    "anchor": "#L1",
    "child": [
      {
        "anchor": "#L2",
        "child": [
          {
            "anchor": "#L3",
            "child": [
              {
                "anchor": "#L4",
                "child": [],
                "title": "L4",
              },
            ],
            "title": "L3",
          },
        ],
        "title": "L2",
      },
    ],
    "title": "L1",
  },
]
`;

exports[`Test toc parse 1`] = `
[
  {
    "anchor": "#1-Markdown语法教程",
    "child": [
      {
        "anchor": "#1.1-标题",
        "child": [],
        "title": "1.1 标题",
      },
      {
        "anchor": "#1.2-字体",
        "child": [],
        "title": "1.2 字体",
      },
      {
        "anchor": "#1.3-无序列表",
        "child": [],
        "title": "1.3 无序列表",
      },
      {
        "anchor": "#1.4-有序列表",
        "child": [],
        "title": "1.4 有序列表",
      },
      {
        "anchor": "#1.5-引用",
        "child": [],
        "title": "1.5 引用",
      },
      {
        "anchor": "#1.6-链接",
        "child": [],
        "title": "1.6 链接",
      },
      {
        "anchor": "#1.7-图片",
        "child": [],
        "title": "1.7 图片",
      },
      {
        "anchor": "#1.8-分割线",
        "child": [],
        "title": "1.8 分割线",
      },
      {
        "anchor": "#1.9-表格",
        "child": [],
        "title": "1.9 表格",
      },
      {
        "anchor": "#1.10-代码块",
        "child": [],
        "title": "1.10 代码块",
      },
    ],
    "title": "1 Markdown语法教程",
  },
  {
    "anchor": "#2-其他语法",
    "child": [
      {
        "anchor": "#2.1-HTML",
        "child": [],
        "title": "2.1 HTML",
      },
      {
        "anchor": "#Github-警告",
        "child": [],
        "title": "Github 警告",
      },
    ],
    "title": "2 其他语法",
  },
]
`;
